components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useRef, useEffect } from "react";
+import Icon from "../Icon/Icon";
+import "./ProgressBar.scss";
+
+type Props = {
+  value: number;
+  onChange: (newProgress: number) => void;
+};
+
+const ProgressBar: React.FC<Props> = ({ value, onChange }) => {
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragStartX, setDragStartX] = useState(0);
+  const [dragStartValue, setDragStartValue] = useState(value);
+  const progressBarRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleMouseUp = () => {
+      setIsDragging(false);
+    };
+
+    window.addEventListener("mouseup", handleMouseUp);
+
+    return () => {
+      window.removeEventListener("mouseup", handleMouseUp);
+    };
+  }, []);
```

Comment:
It is correct usage for event handlers, however I would recommend using the native react.js methods